store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java

```diff
@@ -555,7 +555,7 @@ public void run() {
                                 return;
                             }
                             log.warn("putResultProcess error, info={}", putResultProcess.toString());
-                            putResultProcess.onException();
+                            putResultProcess.doResend();
```

Comment:
if a exception occur in doResend which will also cause infinite resend， there is not chance to reach SKIP status.

Reply:
It is necessary to retry if an exception occurs, otherwise it will cause data loss.